roles/metrics-export.lua

```diff
@@ -315,17 +363,25 @@ local function apply_http(conf)
         end
     end
 
-    for listen, server in pairs(http_servers) do
-        if not enabled[listen] then
-            server.httpd:stop()
-            http_servers[listen] = nil
+    for target, server in pairs(http_servers) do
+        if not enabled[target] then
+            if not server.is_httpd_role then
+                server.httpd:stop()
+            else
+                for path, _ in pairs(server.routes) do
+                    server.httpd:delete(path)
+                end
+            end
```

Comment:
It just a little easier to read.
```suggestion
            if server.is_httpd_role then
                for path, _ in pairs(server.routes) do
                    server.httpd:delete(path)
                end
             else
                server.httpd:stop()
            end
```

Reply:
Edited.